src/signTxOutput.c

```diff
@@ -384,7 +384,14 @@ static void signTxOutput_handleAssetGroupAPDU(uint8_t* wireDataBuffer, size_t wi
 
 		VALIDATE(view_remainingSize(&view) >= MINTING_POLICY_ID_SIZE, ERR_INVALID_DATA);
 		STATIC_ASSERT(SIZEOF(tokenGroup->policyId) == MINTING_POLICY_ID_SIZE, "wrong policy id size");
-		view_memmove(tokenGroup->policyId, &view, MINTING_POLICY_ID_SIZE);
+		uint8_t candidatePolicyId[MINTING_POLICY_ID_SIZE];
+		view_memmove(candidatePolicyId, &view, MINTING_POLICY_ID_SIZE);
+		VALIDATE(0 == subctx->currentAssetGroup || cbor_mapKeyFulfillsCanonicalOrdering(
+		                 tokenGroup->policyId, MINTING_POLICY_ID_SIZE,
+		                 candidatePolicyId, MINTING_POLICY_ID_SIZE), ERR_INVALID_DATA);
+
+		ASSERT(sizeof(tokenGroup->policyId) >= MINTING_POLICY_ID_SIZE);
```

Comment:
We can use static (compile-time) asserts and `SIZEOF` instead of `sizeof` where it is possible, for example:
```
			STATIC_ASSERT(SIZEOF(ctx->auxDataHash) == AUX_DATA_HASH_LENGTH, "wrong auxiliary data hash length");
			view_memmove(ctx->auxDataHash, &view, AUX_DATA_HASH_LENGTH);
```